Sources/StreamChat/Utils/Logger/Logger.swift

```diff
@@ -178,20 +178,26 @@ public enum LogConfig {
     /// Underlying logger instance to control singleton.
     private static var _logger: Logger?
 
+    private static var queue = DispatchQueue(label: "io.getstream.logconfig")
+
     /// Logger instance to be used by StreamChat.
     ///
     /// - Important: Other options in `LogConfig` will not take affect if this is changed.
     public static var logger: Logger {
         get {
-            if let logger = _logger {
-                return logger
-            } else {
-                _logger = Logger(identifier: identifier, destinations: destinations)
-                return _logger!
+            queue.sync {
+                if let logger = _logger {
+                    return logger
+                } else {
+                    _logger = Logger(identifier: identifier, destinations: destinations)
+                    return _logger!
+                }
             }
         }
         set {
-            _logger = newValue
+            queue.async {
```

Comment:
`async(flags: .barrier)`

Reply:
In a serial queue there isn't a point to use async with .barrier i believe. if we change the queue to be concurrent then this suggestion can be applied

Reply:
https://github.com/GetStream/stream-chat-swift/pull/2775#discussion_r1325540582

Reply:
same as above